dataland-frontend/tests/e2e/specs/eu-taxonomy-financials/EditButtonWithMultipleReportingPeriods.ts

```diff
@@ -0,0 +1,85 @@
+import { describeIf } from "@e2e/support/TestUtility";
+import { getKeycloakToken } from "@e2e/utils/Auth";
+import { generateDummyCompanyInformation, prepareUniqueCompany, uploadCompanyViaApi } from "@e2e/utils/CompanyUpload";
+import { admin_name, admin_pw } from "@e2e/utils/Cypress";
+import { addCreditInstitutionDataset, prepareFixture } from "@e2e/utils/EuTaxonomyFinancialsUpload";
+import { FixtureData, getPreparedFixture } from "@sharedUtils/Fixtures";
+
+describeIf(
+  "As a user, I expect to be able to edit datasets with multiple reporting periods",
+  {
+    executionEnvironments: ["developmentLocal", "ci", "developmentCd"],
+    dataEnvironments: ["fakeFixtures"],
+  },
+  function () {
+    let testData: FixtureData<EuTaxonomyDataForFinancials>;
+    const testCompany = prepareUniqueCompany("company-for-testing-edit-button");
+
+    before(function () {
+      const fixtureType = "CompanyInformationWithEuTaxonomyDataForFinancialsPreparedFixtures";
+      prepareFixture<EuTaxonomyDataForFinancials>(fixtureType).then((preparedFixtures) => {
+        testData = getPreparedFixture("company-for-all-types", preparedFixtures);
+      });
+    });
+
+    it("Check whether newly added dataset has Pending status and can be approved by a reviewer", () => {
+      getKeycloakToken(admin_name, admin_pw).then((token: string) => {
+        return uploadCompanyViaApi(token, generateDummyCompanyInformation(testCompany.companyName)).then(
+          (storedCompany): void => {
+            cy.ensureLoggedIn(admin_name, admin_pw);
+            cy.visitAndCheckAppMount(
+              `/companies/${storedCompany.companyId}/frameworks/${DataTypeEnum.EutaxonomyFinancials}/upload`
+            );
+            addTwoDatasetsWithDifferentReportingPeriods(testData.t, storedCompany);
+            testEditDataButton(storedCompany);
+          }
+        );
+      });
+    });
+  }
+);
+
+/**
+ * Adds two new datasets with different reporting periods
+ * @param data the data to fill the form with
+ * @param storedCompany details of the company that was created
+ */
+function addTwoDatasetsWithDifferentReportingPeriods(
+  data: EuTaxonomyDataForFinancials,
+  storedCompany: StoredCompany
+): void {
+  addCreditInstitutionDataset(data, "2022");
+
+  cy.visit("/companies").wait(4000);
+  cy.visitAndCheckAppMount(
+    `/companies/${storedCompany.companyId}/frameworks/${DataTypeEnum.EutaxonomyFinancials}/upload`
+  );
+
+  addCreditInstitutionDataset(data, "2021");
+}
+
+/**
+ * Tests that the item was added and is visible on the QA list
+ * @param storedCompany details of the company that was created
+ */
+function testEditDataButton(storedCompany: StoredCompany): void {
+  cy.visit("/companies").wait(4000);
+
+  cy.get('[data-test="search-result-framework-data"] .p-datatable-tbody')
+    .first()
+    .should("exist")
+    .should("contain", storedCompany.companyInformation.companyName)
+    .click();
```

Comment:
I don't think it is necessary to go to the search page and click on the company you created there in the search results.  This is covered by other tests and just takes time.

You could just visit the view page for lksg directly via its url, which will be:
dataland.com/companies/<companyId>/frameworks/lksg

Reply:
Done.